trin-core/src/jsonrpc/types.rs

```diff
@@ -101,18 +98,55 @@ impl TryFrom<&Value> for PingParams {
     type Error = ValidationError;
 
     fn try_from(param: &Value) -> Result<Self, Self::Error> {
-        match param.as_str() {
-            Some(val) => match Enr::from_str(val) {
-                Ok(enr) => Ok(Self { enr }),
-                Err(_) => Err(ValidationError::new("Invalid enr param")),
+        let enr: SszEnr = match param.try_into() {
+            Ok(val) => val,
+            Err(msg) => return Err(msg),
+        };
```

Comment:
Looks equivalent to:
```suggestion
        let enr: SszEnr = param.try_into()?;
```